app/controllers/dor_controller.rb

```diff
@@ -2,17 +2,21 @@
 
 # Main controller of application
 class DorController < ApplicationController
+  include Dry::Monads[:result]
+
+  class CocinaModelBuildError < StandardError; end
+
   def reindex
-    indexer = Indexer.new(solr: solr, pid: params[:pid])
-    cocina_with_metadata = indexer.fetch_model_with_metadata
-    @solr_doc = indexer.reindex_pid(
-      add_attributes: { commitWithin: params.fetch(:commitWithin, 1000).to_i },
-      cocina_with_metadata: cocina_with_metadata
-    )
-    indexer.commit unless params[:commitWithin] # reindex_pid doesn't commit, but callers of this method may expect the update to be committed immediately
+    @indexer = Indexer.new(solr: solr, pid: params[:pid])
+
+    reindex_pid
     render status: :ok, plain: "Successfully updated index for #{params[:pid]}"
   rescue Dor::Services::Client::NotFoundResponse, Rubydora::RecordNotFound
     render status: :not_found, plain: 'Object does not exist in the repository'
+  rescue CocinaModelBuildError => e
+    request.session # TODO: calling this as a hack to address bad Rails/HB interaction, remove when https://github.com/rails/rails/issues/43922 is fixed
```

Comment:
so i lucked out (?) and found this problem because i wrote my build error test before doing an `allow`/`expect` on the HB notification, and i was perplexed by the fact that i was getting `NoMethodError: undefined method `enabled?' for {}:Hash` on the line below.  after confirming that i was indeed calling `Honeybadger.notify` the same way as i saw it called in our other apps, and after trying (unsuccessfully) to reproduce this weird behavior by getting rid of a similar `allow(Honeybadger).to receive(:notify)` in `spec/services/document_builder_spec.rb` (which exercises `app/indexers/identifiable_indexer.rb`), i did some web searching, and came across https://github.com/rails/rails/issues/43922 (filed by our very own @jcoyne 🙂).

i tried adding this (otherwise extraneous) `request.session` (inspired by [jcoyne's debugging here](https://github.com/rails/rails/issues/43922#issuecomment-999751892)), and that made the unexpected error go away, even when i let my test run the actual `Honeybadger.notify` call without mocking it.

totally open to suggestions for other workarounds for this issue.  the other obvious option from that thread seemed to be monkeypatching `Rack::Session::Abstract::PersistedSecure::SecureSessionHash`, which felt worse to me than what i did here.